lib/active_record/virtual_attributes/virtual_fields.rb

```diff
@@ -202,6 +205,20 @@ def grouped_records(orig_association, records, polymorphic_parent)
             h
           end
           # rubocop:enable Style/BlockDelimiters, Lint/AmbiguousBlockAssociation, Style/MethodCallWithArgsParentheses
+
+          def display_virtual_attribute_deprecation(str, skip)
+            return if skip
+
+            short_caller = caller
+            unless ActiveSupport::Deprecation.debug
```

Comment:
This seems backwards.  Shouldn't this be `if ActiveSupport::Deprecation.debug`, then we add the backtrace?

Reply:
debug says "please print a long stack trace" so in that instance we don't want to filter the backtrace down.

I wanted the backtrace to point people to the spot in the code where they used the bad include.

Reply:
I put in a comment to explain

Reply:
I see.  We always print a stacktrace, but how "big" said trace is is based on this.

Fair enough.

Reply:
the default case is printing 1 line. which it printed a line in virtual attributes which was no help.
now it points the the ruby that called the query. not perfect, but the best we can do. (shortcoming of active record itself)